### PR TITLE
part 2 of Google Actions for Android Devs

### PR DESCRIPTION
**Pull request by patjackson52**

### CODE REVIEW

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+title: "Google Assistant Development for Android Devs (Part 2)"
+author: Patrick Jackson
+categories: Development
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: We have an "Android" category in addition to "Development", so let's file this under both those categories and remove the "Android" tag.

Maybe update the old post as well, sorry I didn't catch this last time. For reference, the list of categories is [here](https://github.com/TicketmasterMobileStudio/ticketmastermobilestudio.github.io#categories-and-tags).

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+---
+
+
+# Google Assistant Development for Android Devs (Part 2)
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Redundant title, can be removed.

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+![](https://developers.google.com/actions/images/surfaces-hero.svg)
+
+This is part 2 in a series on developing Actions on Google.  [Part one is here](http://ticketmastermobilestudio.com/blog/google-assistant-development-for-android-devs-part-1).
+
+In my last post, I introduced an open source SDK for Actions on Google for Kotlin and Java.  Today, that library port is complete, with all features supported.  
+
+ - Api.ai support
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: They stylize it in all caps: API.AI. (Same change for the 2 mentions of API.AI in the bullet list below)

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+ - Actions SDK support
+ - All tests ported and passing (~230)
+ - Conversation components example for Api.Ai & Actions SDK
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: API.AI stylization (all caps)

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    	@Override
+    	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+         	 ApiAiAction action = new ApiAiAction(req, resp);
+       	 	 action.handleRequest(intentMap);
+    	}
+    }
+
+
+ 
+Many Android and Java developers should be able to get started quickly, even if your a newbie to backend development.  Check out [our github repo]((https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin)) for more information on how to be started and examples.
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: - your -> you're
- backend -> back-end (I think?)
- github -> GitHub
- be started -> get started
- Suggest rewording the last sentence to "... for examples and more information on getting started"

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+Next post I'll be covering some of the basic concepts and tools for developing Actions on Google.
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: - Next post -> In the next post

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ - All tests ported and passing (~230)
+ - Conversation components example for Api.Ai & Actions SDK
+ - Transaction Sample for Api.Ai
+ - Java conversation sample
+ 
+Now the capabilities of the [official Node.js SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin) can be matched easily in Kotlin & Java (or any JVM language).  Since this is a port, and the API is nearly identical, one can look at the Node.js docs and examples and easily do the same in Kotlin or Java.  Below is how writing Actions for Google with the unofficial SDK looks.
+
+### Actions in Kotlin
+
+    fun welcome(app: ApiAiApp) =
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Let's stick to the backtick syntax for code blocks, since it allows us to specify language and isn't as prone to mistakes (your second block is not all 4-spaces indented, so it's throwing some errors right now):

    ```kotlin
    // blah blah
    ```

    ```java
    // blah blah
    ```

**Review by yorb — CHANGES_REQUESTED** [review]
Looks good with the minor changes I listed (rejecting only because of the front matter changes, the rest are just suggestions).

Also don't forget to change the date in the filename :)

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 8]
```diff
@@ -1,30 +1,29 @@
 ---
 title: "Google Assistant Development for Android Devs (Part 2)"
 author: Patrick Jackson
-categories: Development
-tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+categories: Development, Android
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Oops sorry, Jekyll wants lists to have brackets if you use commas, so either: `Development Android` or `[Development, Android]`

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 9]
```diff
@@ -1,30 +1,29 @@
...
+tags: [Google Assistant, Actions on Google, conversational interfaces, Kotlin]
 ---
 
 
-# Google Assistant Development for Android Devs (Part 2)
-
 ![](https://developers.google.com/actions/images/surfaces-hero.svg)
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: I wonder if a different graphic would be better here, just to avoid repetition with part 1.

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md (hunk 7 above)
Comment (yorb) [reply _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Much better—can you also take out the extra indentation? Just a nitpick, it just adds unnecessary space to the left side of the code block now that you added the backticks.

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 10]
```diff
@@ -59,9 +58,11 @@ Now the capabilities of the [official Node.js SDK](https://github.com/Ticketmast
         	ApiAiAction(req, resp).handleRequest(actionMap)
        }
     }
+```
 
 ### Actions in Java
 
+```java
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: So this code block is still throwing an error in Liquid (Jekyll's templating engine) because of the double `{{` starting on line 90. The best way I know to avoid it is to wrap the whole code block in Liquid's "raw" tags:

    {% raw %}
      ```java
      // ...
      ```
    {% endraw %}

Ugly I know but without this it's omitting everything between the `{{` and `}}` in your code.

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 11]
```diff
@@ -98,10 +99,11 @@ Now the capabilities of the [official Node.js SDK](https://github.com/Ticketmast
        	 	 action.handleRequest(intentMap);
     	}
     }
+```
 
 
  
-Many Android and Java developers should be able to get started quickly, even if your a newbie to backend development.  Check out [our github repo]((https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin)) for more information on how to be started and examples.
+Many Android and Java developers should be able to get started quickly, even if yor're a newbie to back-end development.  Check out [our GitHub repo]((https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin)) for examples and more information on getting started.
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Typo: yor're -> you're

**Review by yorb — CHANGES_REQUESTED** [review]
Couple more things I noticed - also don't forget the date in the filename.

File: _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md [hunk 12]
```diff
@@ -1,30 +1,29 @@
...
+ - API.AI support
  - Actions SDK support
  - All tests ported and passing (~230)
- - Conversation components example for Api.Ai & Actions SDK
- - Transaction Sample for Api.Ai
+ - Conversation components example for API.AI & Actions SDK
+ - Transaction Sample for API.AI
  - Java conversation sample
  
 Now the capabilities of the [official Node.js SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin) can be matched easily in Kotlin & Java (or any JVM language).  Since this is a port, and the API is nearly identical, one can look at the Node.js docs and examples and easily do the same in Kotlin or Java.  Below is how writing Actions for Google with the unofficial SDK looks.
```
Comment (yorb) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: Can you add `{:target='_blank'}` after this URL so it opens in a new tab? Same with the link in the last paragraph.

Comment (mgibble) [inline _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: putting the link on the text "official Node.js SDK" makes me think you are linking to the official node.js. Since it is a link to your library, might be better to put it with words that ARE your sdk. Might link it on the paragraph above when you reference "open source SDK"?

Comment (patjackson52) [reply _posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md]: oh!  that should go to the node.js lib.

File: _posts/blog/2017-08-21-google-assistant-development-for-android-devs-part-2.md [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+In my last post, I introduced an open source SDK for Actions on Google for Kotlin and Java.  Today, that library port is complete, with all features supported.  
+
+ - API.AI support
+ - Actions SDK support
+ - All tests ported and passing (~230)
+ - Conversation components example for API.AI & Actions SDK
+ - Transaction Sample for API.AI
+ - Java conversation sample
+
```
Comment (yorb) [inline _posts/blog/2017-08-21-google-assistant-development-for-android-devs-part-2.md:21]: I forgot about this, but Jekyll automatically sets the excerpt to be the first paragraph, which in this case is just the image. It might be good to include a bit more before the break, maybe here after the bullet list? You just have to add this bit:

```html
<!--more-->
```

File: _posts/blog/2017-08-21-google-assistant-development-for-android-devs-part-2.md [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+    	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+         	 ApiAiAction action = new ApiAiAction(req, resp);
+       	 	 action.handleRequest(intentMap);
+    	}
+    }
+```
+{% endraw %}
+
+ 
+Many Android and Java developers should be able to get started quickly, even if your're a newbie to back-end development.  Check out [our GitHub repo]((https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin) for examples and more information on getting started.
```
Comment (yorb) [inline _posts/blog/2017-08-21-google-assistant-development-for-android-devs-part-2.md]: - your're -> you're (the typo that wouldn't die! haha)
- You have an extra open paren in the URL, which is making the link break
- Maybe also add a `{:target='_github'}` to this URL and the one earlier up?

**Review by yorb — APPROVED** [review]
Sorry this review is late - last couple things. Looks good after that so I'm approving, you can fix them and then publish. Nice work!
